authenticator/authenticator_test.go

```diff
@@ -30,3 +224,43 @@ func TestXorBytes(t *testing.T) {
 		t.Fatalf("expected %#v, but received %#v", expected, result)
 	}
 }
+
+func setup(t *testing.T) *httptest.Server {
+	// These tests rely upon the file back-end, so unset the Vault addr if it exists.
+	os.Setenv(vault.EnvVaultAddress, "")
+
+	// Now mock the AWS server.
+	defaultArn := "arn:aws:iam::403019568400:assumed-role/dev"
+	unauthorizedArn := "arn:aws:sts::123456789012:federated-user/malicious-user"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		responseTemplate := `<GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
+  <GetCallerIdentityResult>
+    <Arn>%s</Arn>
+    <UserId>123456789012:my-federated-user-name</UserId>
+    <Account>123456789012</Account>
+  </GetCallerIdentityResult>
+  <ResponseMetadata>
+    <RequestId>01234567-89ab-cdef-0123-456789abcdef</RequestId>
+  </ResponseMetadata>
+</GetCallerIdentityResponse>`
```

Comment:
TODO use real AWS.

Reply:
Pushed in https://github.com/approzium/approzium/pull/47/commits/427466dda7cf7c1b3c451e9d440dc34cfd9e0a5d